extension/src/status_manager.ts

```diff
@@ -0,0 +1,54 @@
+import type { GenerateManager } from "@ailly/core/lib/actions/generate_manager.js";
+import type { StatusBarItem } from "vscode";
+import vscode, { StatusBarAlignment } from "vscode";
+
+export interface StatusManager {
+  track(manager: GenerateManager): void;
+}
+
+export class StatusBarStatusManager implements StatusManager {
+  outstanding: number = 0;
+  statusBarItem: StatusBarItem;
+
+  static withContext({ subscriptions }: vscode.ExtensionContext) {
+    const statusManager = new StatusBarStatusManager();
+    subscriptions.push(statusManager.statusBarItem);
+    return statusManager;
+  }
+
+  private constructor() {
+    this.statusBarItem = vscode.window.createStatusBarItem(
+      "ailly.statusBarItem",
+      StatusBarAlignment.Right
+    );
+    this.updateStatusBarItem();
+  }
+
+  track(manager: GenerateManager) {
+    manager.threads.forEach((thread) =>
+      thread.forEach(async (content) => {
+        this.addOutstanding();
+        await content.responseStream.promise;
+        this.finishOutstanding();
+      })
+    );
+  }
+
+  addOutstanding(count: number = 1) {
+    this.outstanding += count;
+    this.updateStatusBarItem();
+  }
+  finishOutstanding() {
+    this.outstanding -= 1;
+    this.updateStatusBarItem();
+  }
+
+  updateStatusBarItem() {
+    this.statusBarItem.text = `Ailly: ${this.outstanding}`;
```

Comment:
Is this useful? As a user I don't know how much I care about active threads. I _do_ care about my VSCode status bar real-estate.

Reply:
Added a (maybe configurable in the future) delay to hide the status bar after getting to 0 outstanding items.